SUS-1215 Be more forgiving of older blocks

garthwebb opened the pull request:
Older blocks may not have the ps_blocker_hit column set in the phalanx_stats table. Existing code was using the null coalescing operator to check both ps_blocker_hit then ps_blocker_type.  However as this value is sometimes zero as well as null, this did not do the right thing. Also add back in an old block for stats display purposes, but hide it for block creation.

https://wikia-inc.atlassian.net/browse/SUS-1215

Note that this PR builds on the PR for SUS-1278 (incorrectly checked in to the branch SOC-1278 but will stay there now, sorry :/).  The PR for that is https://github.com/Wikia/app/pull/11832

@nmonterroso @TK-999

Commit:
SUS-1215 Be more forgiving of older blocks

Older blocks may not have the ps_blocker_hit column set in the phalanx_stats table. Existing code was using the null coalescing operator to check both ps_blocker_hit then ps_blocker_type.  However as this value is sometimes zero as well as null, this did not do the right thing. Also add back in an old block for stats display purposes, but hide it for block creation.

WikiaTech commented on extensions/wikia/PhalanxII/classes/Phalanx.class.php:
```diff
@@ -174,7 +183,12 @@ public function save() {
+				__METHOD__
+			);
 			$action = 'edit';
 		};
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this empty statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1116)

WikiaTech commented:
SonarQube analysis reported 25 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 5 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 15 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxSpecialController.class.php#L295](https://github.com/Wikia/app/blob/adfd3882149e19a2e59bb665de7443702dd93469/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L295): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L5](https://github.com/Wikia/app/blob/adfd3882149e19a2e59bb665de7443702dd93469/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L5): Class "Phalanx" has 24 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/adfd3882149e19a2e59bb665de7443702dd93469/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): The Cyclomatic Complexity of this function "onEditPhalanxBlock" is 28 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/adfd3882149e19a2e59bb665de7443702dd93469/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L104](https://github.com/Wikia/app/blob/adfd3882149e19a2e59bb665de7443702dd93469/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L104): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L13](https://github.com/Wikia/app/blob/adfd3882149e19a2e59bb665de7443702dd93469/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L13): Define a constant instead of duplicating this literal "Phalanx" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L34](https://github.com/Wikia/app/blob/adfd3882149e19a2e59bb665de7443702dd93469/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L34): Replace this "switch" statement with "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L145](https://github.com/Wikia/app/blob/adfd3882149e19a2e59bb665de7443702dd93469/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L145): Define a constant instead of duplicating this literal "checkBlocker" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L308](https://github.com/Wikia/app/blob/adfd3882149e19a2e59bb665de7443702dd93469/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L308): Define a constant instead of duplicating this literal "error" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L315](https://github.com/Wikia/app/blob/adfd3882149e19a2e59bb665de7443702dd93469/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L315): Define a constant instead of duplicating this literal "token" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

garthwebb commented on extensions/wikia/PhalanxII/PhalanxSpecialController.class.php:
```diff
@@ -106,7 +106,7 @@ public function main() {
 		// VSTF should not be allowed to block emails in Phalanx
 		$showEmailBlock = $this->wg->User->isAllowed( 'phalanxemailblock' );
-		$blockTypes  = Phalanx::getAllTypeNames();
+		$blockTypes  = Phalanx::getSupportedTypeNames();
```
We got rid of `128 => 'cookie'` as a block type back in early 2016 but that bit value still lives on in some of our stats.  Make sure there is a way to return the types we use while still having all the types that exist available for stats reporting.

garthwebb commented on extensions/wikia/PhalanxII/classes/Phalanx.class.php:
```diff
@@ -10,6 +10,7 @@ class Phalanx extends WikiaModel implements ArrayAccess {
 	const TYPE_ANSWERS_QUESTION_TITLE = 16;
 	const TYPE_ANSWERS_RECENT_QUESTIONS = 32;
 	const TYPE_WIKI_CREATION = 64;
+	const TYPE_COOKIE = 128;
```
See comment above about adding this back in for use in PhalanxStats.

garthwebb commented on extensions/wikia/PhalanxII/classes/Phalanx.class.php:
```diff
@@ -288,7 +317,10 @@ private function log( $action ) {
+			wfMsgExt(
+				'phalanx-rule-log-details', [ 'parsemag', 'content' ],
+				$this->data['text'], $types, $this->data['reason']
+			)
```
I would have updated this to `wfMessage` but there aren't any clear docs on online to address `content` as an option.

mszabo-wikia replied:
@garthwebb we should use `Message::inContentLanguage` for that, ie: `wfMessage(...)->inContentLanguage()->escaped();`

garthwebb commented on extensions/wikia/PhalanxII/classes/PhalanxStatsPager.class.php:
```diff
@@ -45,17 +45,28 @@ function getPagingQueries() {
 	function formatRow( $row ) {
-		$type = implode( ", ", Phalanx::getTypeNames( $row->ps_blocker_hit ?? $row->ps_blocker_type ) );
+		$blocker = $row->ps_blocker_hit ?: $row->ps_blocker_type;
+		$type = implode( ', ', Phalanx::getTypeNames( $blocker ) );
```
Use the elvis operator (`?:`) rather than the null coalescing operator (`??`) as `ps_blocker_hit` can sometimes be zero which is just as invalid as `null` in this case.

garthwebb commented on extensions/wikia/PhalanxII/classes/PhalanxStatsPager.class.php:
```diff
@@ -45,17 +45,28 @@ function getPagingQueries() {
+
+		$url = $row->ps_referrer ?: '';
+		if ( empty( $url ) ) {
+			$wiki = WikiFactory::getWikiByID( $row->ps_wiki_id );
```
We were doing this wiki factory look up every time previously even if we didn't need it because `$row->ps_referrer` was defined.  Only load wiki information from the DB if necessary.

garthwebb commented on extensions/wikia/PhalanxII/classes/PhalanxStatsPager.class.php:
```diff
@@ -45,17 +45,28 @@ function getPagingQueries() {
+			NS_SPECIAL,
+			$row->ps_wiki_id
+		);
+		$specialContributionsURL = $specialContributions->getFullURL();
```
Efforts to get our line lengths down.

Commit:
SUS-1215 Don't create a URL if the server is not present.

The GlobalTitle will create a URL even the server name is not available, leading to a broken URL.

Also update the schema to reflect reality.

WikiaTech commented:
SonarQube analysis reported 25 issues
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 5 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 15 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxSpecialController.class.php#L295](https://github.com/Wikia/app/blob/d6f072ddd46bf06ce22a66ad4e407bd6d64f2313/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L295): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L5](https://github.com/Wikia/app/blob/d6f072ddd46bf06ce22a66ad4e407bd6d64f2313/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L5): Class "Phalanx" has 24 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/d6f072ddd46bf06ce22a66ad4e407bd6d64f2313/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): The Cyclomatic Complexity of this function "onEditPhalanxBlock" is 28 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/d6f072ddd46bf06ce22a66ad4e407bd6d64f2313/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L104](https://github.com/Wikia/app/blob/d6f072ddd46bf06ce22a66ad4e407bd6d64f2313/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L104): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L13](https://github.com/Wikia/app/blob/d6f072ddd46bf06ce22a66ad4e407bd6d64f2313/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L13): Define a constant instead of duplicating this literal "Phalanx" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L34](https://github.com/Wikia/app/blob/d6f072ddd46bf06ce22a66ad4e407bd6d64f2313/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L34): Replace this "switch" statement with "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L145](https://github.com/Wikia/app/blob/d6f072ddd46bf06ce22a66ad4e407bd6d64f2313/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L145): Define a constant instead of duplicating this literal "checkBlocker" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L308](https://github.com/Wikia/app/blob/d6f072ddd46bf06ce22a66ad4e407bd6d64f2313/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L308): Define a constant instead of duplicating this literal "error" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
1. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L315](https://github.com/Wikia/app/blob/d6f072ddd46bf06ce22a66ad4e407bd6d64f2313/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L315): Define a constant instead of duplicating this literal "token" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)

nmonterroso commented:
LGTM

Commit:
SUS-1215 Update wfMsgExt to wfMessasge(...)

WikiaTech commented:
SonarQube analysis reported 25 issues
- ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 5 major
- ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 15 minor
- ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxSpecialController.class.php#L295](https://github.com/Wikia/app/blob/215831bcbe7f5b3c309565010ec5d4bc17bc0a9e/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L295): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
2. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [Phalanx.class.php#L5](https://github.com/Wikia/app/blob/215831bcbe7f5b3c309565010ec5d4bc17bc0a9e/extensions/wikia/PhalanxII/classes/Phalanx.class.php#L5): Class "Phalanx" has 24 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
3. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/215831bcbe7f5b3c309565010ec5d4bc17bc0a9e/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): The Cyclomatic Complexity of this function "onEditPhalanxBlock" is 28 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
4. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L93](https://github.com/Wikia/app/blob/215831bcbe7f5b3c309565010ec5d4bc17bc0a9e/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L93): Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)
5. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [PhalanxHooks.class.php#L104](https://github.com/Wikia/app/blob/215831bcbe7f5b3c309565010ec5d4bc17bc0a9e/extensions/wikia/PhalanxII/hooks/PhalanxHooks.class.php#L104): Add curly braces around the nested statement(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
6. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L13](https://github.com/Wikia/app/blob/215831bcbe7f5b3c309565010ec5d4bc17bc0a9e/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L13): Define a constant instead of duplicating this literal "Phalanx" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
7. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L34](https://github.com/Wikia/app/blob/215831bcbe7f5b3c309565010ec5d4bc17bc0a9e/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L34): Replace this "switch" statement with "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1301)
8. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L145](https://github.com/Wikia/app/blob/215831bcbe7f5b3c309565010ec5d4bc17bc0a9e/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L145): Define a constant instead of duplicating this literal "checkBlocker" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
9. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L308](https://github.com/Wikia/app/blob/215831bcbe7f5b3c309565010ec5d4bc17bc0a9e/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L308): Define a constant instead of duplicating this literal "error" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
10. ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) [PhalanxSpecialController.class.php#L315](https://github.com/Wikia/app/blob/215831bcbe7f5b3c309565010ec5d4bc17bc0a9e/extensions/wikia/PhalanxII/PhalanxSpecialController.class.php#L315): Define a constant instead of duplicating this literal "token" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
